vaapi/vaapiimagepool.cpp

```diff
@@ -69,6 +69,7 @@ struct VaapiImagePool::ImageRecycler
         if (!image)
             return;
         m_pool->recycleID(image->getID());
+        m_pool.reset();
```

Comment:
Hi mdnavare:
This patch can fix VaapiDisplay leak problem, but i still see the random failed.Can you test more times?
I find the reason of this leak, I will submit other patch and explain details in it